### ISSUE TITLE
addAttachment() results in RallyRESTAPIError (422 Could not convert ...)

### ISSUE THREAD

**Issue by rmcfatter** [RallyTools/RallyRestToolkitForPython#46]
The Base64 encoding used by pyral can result in the Rally server rejecting attachments:

    File "/usr/local/lib/python2.7/dist-packages/pyral/restapi.py", line 970, in put
        raise RallyRESTAPIError('%s %s' % (response.status_code, desc))
    pyral.restapi.RallyRESTAPIError: 422 Could not convert:  : value to convert :

This appears to be resolved by switching from the base64 "legacy" interface to the Python > 2.4 API.  At restapi.py:1309:

```
@@ -1306,8 +1306,8 @@ class Rally(object):
 
         contents = ''
         with open(filename, 'r') as af:
-            contents = base64.encodestring(af.read())
-            
+            contents = base64.b64encode(af.read())
+
```

**Comment by klehman-rally** [comment]
Great catch, thank you!
 This'll go into pyral 1.1.1.

**Comment by klehman-rally** [comment]
Fixed in 1.1.1. The origin of changed behavior was a change in the back end processing of WSAPI v2.0 requests on the Rally side.
